modules/asset-manager-impl/src/main/java/org/opencastproject/assetmanager/impl/AssetManagerImpl.java

```diff
@@ -940,7 +945,13 @@ public IndexRebuildService.Service getService() {
   }
 
   @Override
-  public void repopulate() throws IndexRebuildException {
+  public void repopulate(String type) throws IndexRebuildException {
```

Comment:
This spits out an NPE when you trigger a rebuild using `http://localhost:8080/index/rebuild`.  This is due to [this line](https://github.com/opencast/opencast/pull/5056/files#diff-3cb53a562e43f510142236fdfc62041d78ebfdb445656a1c99f187de65738b08R156), which is passing a `null` which isn't properly handled here.

This should read
```suggestion
  public void repopulate(String type) throws IndexRebuildException {
    ServicePart parsedType = null;
    if (type != null && EnumUtils.isValidEnum(ServicePart.class, type)) {
      parsedType = ServicePart.valueOf(type);
    else {
      throw new IndexRebuildException("The given type " + type + " was not valid. Should be null or "
          + ServicePart.values());
    }
```

Caution: Above *probably* has spacing issues, this UI is not helpful with multiline suggestions.  Also don't forget the next suggestion below!